### PR TITLE
build: keep specific z version in go mod 1.22.5

### PR DESCRIPTION
**Pull request by subhamkrai**
keep specific z version in go mod 1.22.5

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by subhamkrai** [comment]
@Nikhil-Ladha

**Review by Madhu-1 — APPROVED** [review]

**Comment by subhamkrai** [comment]
@Madhu-1 when running make build it is reverting the change `go 1.22.0` to `go 1.22.5` seems like we can do this change

**Comment by subhamkrai** [comment]
https://github.com/rook/rook/actions/runs/11013022253/job/30580437404?pr=14761#step:5:31

**Comment by Madhu-1** [comment]
We can have 1.22.5 as the minimum version as csi-addons also as minimal support

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/rook/rook
 
-go 1.22.6
+go 1.22.5
```
Comment (parth-gr) [inline go.mod:3]: I see a different version in toolchain

**Review by parth-gr — CHANGES_REQUESTED** [review]

File: pkg/apis/go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/rook/rook/pkg/apis
 
-go 1.22.6
+go 1.22.0
```
Comment (Madhu-1) [inline pkg/apis/go.mod:3]: it should be 1.22.5 atleast due to csi-addons

File: go.mod (hunk 1 above)
Comment (subhamkrai) [reply go.mod:3]: toolchain version will be latest z version

File: pkg/apis/go.mod (hunk 2 above)
Comment (subhamkrai) [reply pkg/apis/go.mod:3]: in that case we should need this pr right? @Madhu-1 as we are already in 1.22.6?

Comment (Madhu-1) [reply pkg/apis/go.mod:3]: Yes, we need to use 1.22.5 as the minimal required version

Comment (subhamkrai) [reply pkg/apis/go.mod:3]: > in that case we should need this pr right? @Madhu-1 as we are already in 1.22.6?

edit: `not` need

File: go.mod (hunk 1 above)
Comment (parth-gr) [reply go.mod:3]: >remove specific z version in go mod

The point was to remove the Z version, I am not following it 
Removing mine needinfo and can talk separately

Comment (subhamkrai) [reply go.mod:3]: please check the slack thread for more details

**Review by parth-gr — APPROVED** [review]

**Comment by parth-gr** [comment]
> let's not keep specific z version in go.mod as it will be taken from toolchain otherwise it will casue problem.
> 
> **Checklist:**
> 
> * [ ]  **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
> * [ ]  Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
> * [ ]  [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
> * [ ]  Documentation has been updated, if necessary.
> * [ ]  Unit tests have been added, if necessary.
> * [ ]  Integration tests have been added, if necessary.

Update the commit message

File: pkg/apis/go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/rook/rook/pkg/apis
 
-go 1.22.6
+go 1.22.5
```
Comment (parth-gr) [inline pkg/apis/go.mod:3]: should we run `go mod tidy` ?

Comment (subhamkrai) [reply pkg/apis/go.mod:3]: I have run already otherwise build CI would have failed

Comment (parth-gr) [reply pkg/apis/go.mod:3]: Okay, I see rook is utilizing the CSI addons API,
And it is missing in the rook API folder, which is correct.....

**Review by parth-gr — APPROVED** [review]

**Pull request merged**